Visualizing and understanding data blog post (#215)
readthedocs/ethicalads.io: content/posts/2021-visualizing-ad-data-at-ethicalads.md

Title: Visualizing Ad Data at EthicalAds
Date: November 16, 2021
description: Data exploration, key performance indicators, and helping our customers all required us to put together a data stack to better understand our ads
tags: graphing, business, kpis
authors: David Fischer
image: /images/posts/2021-graphing-kpis.jpg
image_credit: <span>Photo by <a href="https://unsplash.com/@kmuza?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Carlos Muza</a> on <a href="https://unsplash.com/photos/hpjSkU2UYSU?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText">Unsplash</a></span>

> While we don't track or store personal data at EthicalAds,

Reviewer: This doesn't hook me in for whatever reason. It sort of downplays this important policy of ours. How about 
```suggestion
Respecting user privacy doesn't mean flying blind. 
While we don't track or store personal data at EthicalAds,
```

running an ad network does generate a ton of data!

> This is data like the publishers where certain ads perform best

Reviewer: These are all insights you gleam from data, not what we're actually storing and reporting. Perhaps listing the actual data on which ads were shown when, whether or not they were clicked, etc. then explaining how we are using this data to extract insights like which publishers perform the best or what's the best time of day to run certain ads.

Author: What do you think of something like this:

```
Respecting user privacy doesn't mean flying blind. 
While we don't track or store personal data at EthicalAds,
running an ad network does generate a ton of data:

- When and where (country) the ad was served
- Whether the ad was viewed and clicked
- The publisher on whose site the ad was served
- The context of why the ad was chosen (it's a data science ad on a publisher about machine learning)

This helps us gain insights like the publishers where certain ads perform best
or the best time of day or region to run certain ads.
```

or the best time of day or region to run certain ads.

> However, the data was a bit hard to explore,

Reviewer: Why?

Author: Perhaps I could give more details but the next sentence does deal with that ("It was tied up in Postgres and usually required complex joins...").

let alone creating and telling a compelling story from the data.
It was tied up in PostgreSQL and usually required complex joins

> to get at anything meaningful.

Reviewer: What do you mean by meaningful? This is where explaining the distinction between what data we actually had vs what insights we were hoping to gleam comes in.

> To make a long story short, we knew the data we collected was important,

Reviewer: I like explaining how it's important rather than making the statement that it is and you should just get why. 
```suggestion
We knew the data we were collecting could help us understand and improve our product,
```

but exploring our data, understanding it,
and presenting it to our advertisers, publishers, or even just ourselves was hard.

## Business intelligence - more than just an oxymoron

> We knew we needed a tool to help.

Reviewer: ```suggestion
We knew we needed a tool to bridge the gap between what we had and what we wanted to know.
```

Ideally, it could help us with exploration, tracking our key performance indicators,
and presenting data to our advertisers, publishers, and ourselves.

We gave [Metabase](https://www.metabase.com/?ref=ethicalads.io) a try

> on a recommendation from [Eric Berry](https://twitter.com/coderberry),

Reviewer: ```suggestion
on a glowing recommendation from [Eric Berry](https://twitter.com/coderberry),
```

who previously founded CodeFund, another company dedicated to ethical advertising.
Eric raved about Metabase and used it in CodeFund's product before it shut down.

> Maybe as developers we have an implicit bias against data analysis and BI tools

Reviewer: This introduces that you had conflicting feelings with employing analytics which is never stated. I don't dislike it, but I would explicitly mention. "I was skeptical but saw the value in the avoidance of reinventing the wheel."  or something to make this line make more sense

or maybe it's just me and I hadn't found one I liked.
However, after trying Metabase, I was hooked.
You could explore data with or without SQL, share graphs with coworkers,
and there was an easy and secure way to embed them into or product.
Lastly, this never impacts production since our setup exclusively connects to a read-replica.

<div class="postimage text-center">
  <img class="w-75" src="{static}../images/posts/2021-dashboard-screenshot-graphs.png" alt="A screenshot of the EthicalAds dashboard with an embedded Metabase graph">
  <p>The EthicalAds dashboard with an embedded Metabase graph</p>
</div>

In about a week, we went from no data stack at all,
to taking visualizations from the exploration stage right into production.
This is the perfect tool for a small team like ours.

## Data we have yet to explore

A few months back, we started collecting data on how long an ad is on the screen (we call it "view time").
We think this is a really valuable metric in understanding what leads to clickthroughs
and we think that advertisers will really like it.
Advertisers routinely tell us they love all the data we share with them

> and that other networks are far less transparent.

Reviewer: Not sure what we're trying to say with this. It is kind of like sequel baiting for another 'Our ad network is the most transparent' article but I'm not sure.

Author: That's fair. It's a bit of a throwaway.
